plugins/kubernetes/src/components/ErrorPanel/ErrorPanel.tsx

```diff
@@ -26,10 +30,30 @@ const clustersWithErrorsToErrorMessage = (
     return (
       <div key={i}>
         <Typography variant="body2">{`Cluster: ${c.cluster.name}`}</Typography>
-        {c.errors.map((e, j) => {
+        {c.errors.map((kubernetesFetchError, j) => {
+          let message: string;
+          switch (kubernetesFetchError.errorType) {
+            case 'TIMED_OUT': {
+              const { address, port } =
+                kubernetesFetchError as KubernetesTimeoutError;
+              message = `Timed out attempting to connect to Kubernetes API at address '${address}' on port ${port}`;
+              break;
+            }
+            case 'ADDR_NOT_FOUND': {
+              const { hostname } =
+                kubernetesFetchError as KubernetesAddrNotFoundError;
+              message = `Could not resolve Kubernetes API hostname: '${hostname}'`;
+              break;
+            }
+            default:
+              message =
+                'message' in kubernetesFetchError
+                  ? `Error communicating with Kubernetes: ${kubernetesFetchError.errorType}, message: ${kubernetesFetchError.message}`
+                  : `Error fetching Kubernetes resource: '${kubernetesFetchError.resourcePath}', error: ${kubernetesFetchError.errorType}, status code: ${kubernetesFetchError.statusCode}`;
+          }
```

Comment:
i'd suggest breaking out all of this into a helper function (maybe even in this same file), so the code below ends up as `{kubernetesErrorToMessage(kubernetesFetchError)}`. Then the function can use `return` on each case instead ofd assigning to a temp variable